Snowflake tables with a GEOGRAPHY column cannot be explored

**Describe the bug**
Since June 2020, Snowflake introduced a GEOGRAPHY data type: https://docs.snowflake.com/en/sql-reference/data-types-geospatial.html. The feature is in [public preview](https://docs.snowflake.com/en/release-notes/preview-features.html), but I've seen quite a few organizations (including where I work) using it on production and it seems quite stable. Other BI tools (including Tableau) already announced that they are currently working on adding support to it.

Trying to use Snowflake tables in Metabase that have a column of type GEOGRAPHY causes an error:

![image](https://user-images.githubusercontent.com/196840/96244076-defbda80-0fa5-11eb-88cb-9c8d07ce2b98.png)


**Logs**
(provided below, as a comment)

**To Reproduce**
Steps to reproduce the behavior:
1. Create a table on Snowflake with supported data types:
```
create or replace table dbt_fernando.public.metabase_bug (
    a string,
    b number,
    c boolean,
    d timestamp_ntz
);

insert into dbt_fernando.public.metabase_bug values
    ('1', 1, true, current_date());
```

2. Set up the database on Metabase
3. Test. Everything works as expected.

4. Modify the table and include a GEOGRAPHY column
```
create or replace table dbt_fernando.public.metabase_bug (
    a string,
    b number,
    c boolean,
    d timestamp_ntz,
    e geography
);

insert into dbt_fernando.public.metabase_bug values
    ('1', 1, true, current_date(), 'POINT(0 0)');
```

5. Delete the database configuration on Metabase and create it again.

6. I can't explore the database:

![image](https://user-images.githubusercontent.com/196840/96246830-7f073300-0fa9-11eb-8e34-d681a9100c7f.png)


**Expected behavior**

Worst case scenario, I would expect that unsupported column types would be hidden/removed and I would still be able to see the other columns. Another option would be to treat it as a variant (like [here](https://github.com/metabase/metabase/blob/master/modules/drivers/snowflake/src/metabase/driver/snowflake.clj#L114)). That's what the [sqlserver driver does](https://github.com/metabase/metabase/blob/master/modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj#L52).


**Screenshots**
(provided above)

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, Safari 11.1, …)
- Your operating system: (e.x. OS X 10.10, Windows 10.1809, Ubuntu 16.04, …)
- Your databases: Snowflake
- Metabase version: (e.x. 0.19.3)
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Jar-file on Ubuntu, …)
- Metabase internal database: (e.x. H2 (default), Postgres or MySQL)

```
    "version": {
      "date": "2020-07-31",
      "tag": "v0.36.2",
      "branch": "release-0.36.x",
      "hash": "13f0225"
    },
```

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
Note: the more honest and specific you are here the more we will take you seriously.

**Additional context**
Add any other context about the problem here.

I managed to get access to the logs. Triggering a database schema sync causes:

```
java.sql.SQLException: An SQLException was provoked by the following failure: java.lang.IllegalArgumentException: No enum constant net.snowflake.client.jdbc.SnowflakeType.GEOGRAPHY
	at com.mchange.v2.sql.SqlUtils.toSQLException(SqlUtils.java:118)
	at com.mchange.v2.sql.SqlUtils.toSQLException(SqlUtils.java:77)
	at com.mchange.v2.sql.SqlUtils.toSQLException(SqlUtils.java:74)
	at com.mchange.v2.c3p0.impl.NewPooledConnection.handleThrowable(NewPooledConnection.java:505)
	at com.mchange.v2.c3p0.impl.NewProxyResultSet.next(NewProxyResultSet.java:699)
	at clojure.java.jdbc$result_set_seq$thisfn__19043.invoke(jdbc.clj:582)
	at clojure.java.jdbc$result_set_seq$thisfn__19043$fn__19044.invoke(jdbc.clj:583)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:713)
	at clojure.core$next__5386.invokeStatic(core.clj:64)
	at clojure.core$some.invokeStatic(core.clj:2701)
	at clojure.core$some.invoke(core.clj:2692)
	at metabase.driver.sql_jdbc.sync$fields_metadata.invokeStatic(sync.clj:176)
	at metabase.driver.sql_jdbc.sync$fields_metadata.doInvoke(sync.clj:170)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at metabase.driver.sql_jdbc.sync$describe_table_fields.invokeStatic(sync.clj:194)
	at metabase.driver.sql_jdbc.sync$describe_table_fields.doInvoke(sync.clj:188)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at metabase.driver.snowflake$eval1726$fn__1727.invoke(snowflake.clj:251)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.fetch_metadata$fn__52959$table_metadata__52964$fn__52965.invoke(fetch_metadata.clj:18)
	at metabase.sync.fetch_metadata$fn__52959$table_metadata__52964.invoke(fetch_metadata.clj:15)
	at metabase.sync.sync_metadata.fields.fetch_metadata$fn__53281$db_metadata__53286$fn__53287.invoke(fetch_metadata.clj:86)
	at metabase.sync.sync_metadata.fields.fetch_metadata$fn__53281$db_metadata__53286.invoke(fetch_metadata.clj:82)
	at metabase.sync.sync_metadata.fields$fn__53763$sync_fields_for_table_BANG___53772$fn__53775$fn__53776.invoke(fields.clj:75)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:149)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
	at metabase.sync.sync_metadata.fields$fn__53763$sync_fields_for_table_BANG___53772$fn__53775.invoke(fields.clj:74)
	at metabase.sync.sync_metadata.fields$fn__53763$sync_fields_for_table_BANG___53772.invoke(fields.clj:68)
	at clojure.core$partial$fn__5839.invoke(core.clj:2624)
	at clojure.core$map$fn__5866.invoke(core.clj:2753)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clojure.core$filter$fn__5893.invoke(core.clj:2809)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.boundedLength(RT.java:1792)
	at clojure.lang.RestFn.applyTo(RestFn.java:130)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.sync.sync_metadata.fields$fn__53801$sync_fields_BANG___53806$fn__53807.invoke(fields.clj:88)
	at metabase.sync.sync_metadata.fields$fn__53801$sync_fields_BANG___53806.invoke(fields.clj:80)
	at metabase.sync.util$fn__42218$run_step_with_metadata__42223$fn__42227$fn__42229.invoke(util.clj:345)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:102)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:96)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:119)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:116)
	at metabase.sync.util$fn__42218$run_step_with_metadata__42223$fn__42227.invoke(util.clj:342)
	at metabase.sync.util$fn__42218$run_step_with_metadata__42223.invoke(util.clj:337)
	at metabase.sync.util$fn__42410$run_sync_operation__42415$fn__42416$fn__42417.invoke(util.clj:430)
	at clojure.core$mapv$fn__8445.invoke(core.clj:6912)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$reduce.invokeStatic(core.clj:6827)
	at clojure.core$mapv.invokeStatic(core.clj:6903)
	at clojure.core$mapv.invoke(core.clj:6903)
	at metabase.sync.util$fn__42410$run_sync_operation__42415$fn__42416.invoke(util.clj:430)
	at metabase.sync.util$fn__42410$run_sync_operation__42415.invoke(util.clj:424)
	at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480$fn__54481$fn__54482.invoke(sync_metadata.clj:51)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:149)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:145)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:142)
	at clojure.core$partial$fn__5839.invoke(core.clj:2623)
	at metabase.driver$fn__22445.invokeStatic(driver.clj:540)
	at metabase.driver$fn__22445.invoke(driver.clj:540)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__42130.invoke(util.clj:138)
	at metabase.sync.util$with_db_logging_disabled$fn__42127.invoke(util.clj:129)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:102)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:96)
	at metabase.sync.util$with_start_and_finish_logging$fn__42116.invoke(util.clj:114)
	at metabase.sync.util$with_sync_events$fn__42111.invoke(util.clj:88)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__42102.invoke(util.clj:67)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:164)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:161)
	at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480$fn__54481.invoke(sync_metadata.clj:50)
	at metabase.sync.sync_metadata$fn__54475$sync_db_metadata_BANG___54480.invoke(sync_metadata.clj:47)
	at metabase.api.database$fn__55211$fn__55213.invoke(database.clj:614)
	at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: No enum constant net.snowflake.client.jdbc.SnowflakeType.GEOGRAPHY
	at java.base/java.lang.Enum.valueOf(Unknown Source)
	at net.snowflake.client.jdbc.SnowflakeType.valueOf(SnowflakeType.java:24)
	at net.snowflake.client.jdbc.SnowflakeType.fromString(SnowflakeType.java:54)
	at net.snowflake.client.jdbc.SnowflakeUtil.extractColumnMetadata(SnowflakeUtil.java:171)
	at net.snowflake.client.jdbc.SnowflakeDatabaseMetaData$4.next(SnowflakeDatabaseMetaData.java:1925)
	at com.mchange.v2.c3p0.impl.NewProxyResultSet.next(NewProxyResultSet.java:685)
	... 84 more
```

--- a/modules/drivers/snowflake/project.clj
+++ b/modules/drivers/snowflake/project.clj
@@ -2,7 +2,7 @@
   :min-lein-version "2.5.0"
 
   :dependencies
-  [[net.snowflake/snowflake-jdbc "3.12.7"]]
+  [[net.snowflake/snowflake-jdbc "3.12.13"]]
 
   :profiles
   {:provided

--- a/modules/drivers/snowflake/src/metabase/driver/snowflake.clj
+++ b/modules/drivers/snowflake/src/metabase/driver/snowflake.clj
@@ -101,6 +101,7 @@
     :CHARACTER                  :type/Text
     :STRING                     :type/Text
     :TEXT                       :type/Text
+    :GEOGRAPHY                  :type/SerializedJSON
     :BINARY                     :type/*
     :VARBINARY                  :type/*
     :BOOLEAN                    :type/Boolean

